src/host/inputBuffer.cpp

```diff
@@ -905,76 +826,44 @@ bool InputBuffer::_CanCoalesce(const KeyEvent& a, const KeyEvent& b) const noexc
-    const IInputEvent* const pLastStoredEvent = _storage.back().get();
-    if (pFirstInEvent->EventType() == InputEventType::KeyEvent &&
-        pLastStoredEvent->EventType() == InputEventType::KeyEvent)
+    auto& lastEvent = _storage.back();
+
+    if (lastEvent->EventType() == InputEventType::MouseEvent && inEvent->EventType() == InputEventType::MouseEvent)
     {
-        const auto pInKeyEvent = static_cast<const KeyEvent* const>(pFirstInEvent);
-        const auto pLastKeyEvent = static_cast<const KeyEvent* const>(pLastStoredEvent);
+        const auto& inMouse = *static_cast<const MouseEvent*>(inEvent.get());
+        auto& lastMouse = *static_cast<MouseEvent*>(lastEvent.get());
 
-        if (pInKeyEvent->IsKeyDown() &&
-            pLastKeyEvent->IsKeyDown() &&
-            !IsGlyphFullWidth(pInKeyEvent->GetCharData()) &&
-            _CanCoalesce(*pInKeyEvent, *pLastKeyEvent))
+        if (lastMouse.IsMouseMoveEvent() && inMouse.IsMouseMoveEvent())
         {
-            // increment repeat count
-            const auto pKeyEvent = static_cast<KeyEvent* const>(_storage.back().release());
-            WORD repeatCount = pKeyEvent->GetRepeatCount() + pInKeyEvent->GetRepeatCount();
-            pKeyEvent->SetRepeatCount(repeatCount);
-            std::unique_ptr<IInputEvent> tempPtr(pKeyEvent);
-            tempPtr.swap(_storage.back());
-
-            inEvents.pop_front();
+            lastMouse.SetPosition(inMouse.GetPosition());
             return true;
         }
     }
-    return false;
-}
-
-// Routine Description:
-// - Handles records that suspend/resume the console.
-// Arguments:
-// - records - records to check for pause/unpause events
-// Return Value:
-// - None
-// Note:
-// - The console lock must be held when calling this routine.
-// - will throw exception on error
-void InputBuffer::_HandleConsoleSuspensionEvents(_Inout_ std::deque<std::unique_ptr<IInputEvent>>& inEvents)
-{
-    auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
-
-    std::deque<std::unique_ptr<IInputEvent>> outEvents;
-    while (!inEvents.empty())
-    {
-        auto currEvent = std::move(inEvents.front());
-        inEvents.pop_front();
-        if (currEvent->EventType() == InputEventType::KeyEvent)
+    else if (lastEvent->EventType() == InputEventType::KeyEvent && inEvent->EventType() == InputEventType::KeyEvent)
+    {
+        const auto& inKey = *static_cast<const KeyEvent*>(inEvent.get());
+        auto& lastKey = *static_cast<KeyEvent*>(lastEvent.get());
+
+        if (lastKey.IsKeyDown() && inKey.IsKeyDown() &&
+            (lastKey.GetVirtualScanCode() == inKey.GetVirtualScanCode() || WI_IsFlagSet(inKey.GetActiveModifierKeys(), NLS_IME_CONVERSION)) &&
+            lastKey.GetCharData() == inKey.GetCharData() &&
+            lastKey.GetActiveModifierKeys() == inKey.GetActiveModifierKeys() &&
+            // TODO: This behavior is an import from old conhost v1 and has been broken for decades.

```

Comment:
track with issue#?

Reply:
If you don't mind, I don't want to track that separately in my backlog. After all, this is a bug that is directly standing in my path of removing `IsGlyphFullWidth` everywhere that is outside of `TextBuffer` (and `ROW`) and thus would get addressed by me anyways in the near term.

Reply:
Oh, right I could've done that... 😮
But... Eh, I don't want to wait 30min for the CI. I'll just cross my fingers to not die within the next few months. 😄